modules/dataframe/src/datastructures/dataframe.cpp

```diff
@@ -161,6 +161,19 @@ DataFrame::DataFrame(const DataFrame &df) {
     }
 }
 
+DataFrame &DataFrame::operator=(const DataFrame &df) {
+    for (const auto &col : df.columns_) {
```

Comment:
Need to `clear` `columns_` on `this` first. 
Also, need to check such that `this != *df`